### PR TITLE
Update .gitignore

### LINKED ISSUE #78: Clean up .gitignore
There are some entries in the `.gitignore` that are redundant or could be rearranged and labeled correctly.

We should do this after the pending PRs, with changes to the `.gitignore`.

### PR DESCRIPTION
**Pull request by Tobias321**
fixes #78

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,115 +1,53 @@
...
-/android/libs/x86_64/
-/android/gen/
-.idea/
-*.ipr
-*.iws
-*.iml
-/android/out/
-com_crashlytics_export_strings.xml
+## Ant, Gradle and Maven
+assets
```
Comment (cagix) [inline .gitignore:15]: add a trailing slash `/` to indicate directories, i.e. `assets/`... (but for directories only)

File: .gitignore [hunk 2]
```diff
@@ -1,115 +1,53 @@
...
+build
+lib
+libs
+target
+.gradle
```
Comment (cagix) [inline .gitignore:20]: should we put all `.`-pattern to the beginning of the respective sections?

File: .gitignore [hunk 3]
```diff
@@ -1,115 +1,53 @@
...
+.loadpath
 .project
 .metadata/
-/android/bin/
-/core/bin/
-/desktop/bin/
-/html/bin/
-/ios/bin/
-/ios-moe/bin/
+.settings/
```
Comment (cagix) [inline .gitignore:29]: should we sort alphabetically?

File: .gitignore [hunk 4]
```diff
@@ -1,115 +1,53 @@
...
+.externalToolBuilders/
 *.tmp
 *.bak
 *.swp
-*~.nib
-.settings/
-.loadpath
-.externalToolBuilders/
```
Comment (cagix) [inline .gitignore:52]: not sure: do we really don't need this for eclipse?

**Review by cagix — CHANGES_REQUESTED** [review]
👍 this looks promising :)

a few suggestions:
- should we remove all ios related entries?
- should we put all `.` entries to the beginning of the respective sections?
- should we sort all entries in the respective sections alphabetically?
- should we add a trailing `/` to all directory entries? this would make it easier to distinguish files and directories.

**Comment by Tobias321** [comment]
> 👍 this looks promising :)
> 
> a few suggestions:
> 
> * should we remove all ios related entries?
> * should we put all `.` entries to the beginning of the respective sections?
> * should we sort all entries in the respective sections alphabetically?
> * should we add a trailing `/` to all directory entries? this would make it easier to distinguish files and directories.

Thanks, all points seem to be meaningful to me. 🙂 I'll take that into account.

**Comment by Tobias321** [comment]
Please have a re-look. I've sorted the entries programmatically. :)

**Review by cagix — APPROVED** [review]
👍

**Pull request merged**